src/main/java/com/hermes/recruitment/repository/ICandidatRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.hermes.recruitment.repository;
+
+import com.hermes.recruitment.model.Candidat;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public interface ICandidatRepository extends JpaRepository<Candidat, Long> {
+
+    default List<Candidat> findAllBySurname(String surname) {

```

Comment:
we are in JPA, no need to write a method body for this, the JPA knows by itself how to implement this method only by its signature
Besides that, we fetch from the database just the values we need, we shouldn't do any filtering Java side